app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -334,9 +334,13 @@ class BindableAdapter<T : Any> internal constructor(
         object : BindableViewHolder<T>(binding.root) {
           override fun bind(data: T) {
             setViewModel(binding, transformViewModel(data))
-            // Attach lifecycleOwner after viewModel has been attached data to the view.
-            // Attaching lifecycleOwner before view model initialization can sometimes cause a NullPointerException because data might not be attached to the views yet.
-            binding.lifecycleOwner = getLifecycleOwner()
+            // Attaching lifecycleOwner before view model initialization can sometimes cause a
+            // NullPointerException because data might not be attached to the views yet.
+            // Skip attaching Lifecycle when StateFragment is created as this is causing NPE, yet to find fix.
+            val currentFragment = getLifecycleOwner()?.javaClass?.simpleName.toString()
+            if (currentFragment != "QuestionPlayerFragment" && currentFragment != "StateFragment") {
```

Comment:
We definitely don't want to have special casing like this (plus, it won't work in alpha builds since class names are obfuscated randomly by Proguard). I think we need to figure out why the NPE is happening and fix it.

Reply:
Hey @BenHenning checking on this issue although i'm not able to get to its root.